### PR TITLE
Split Jdifference in dDifference_dq<0> and dDifference_dq<1>

### PR DESCRIPTION
**Pull request by jmirabel**

### CODE REVIEW

**Comment by jcarpent** [comment]
Thanks for this contribution. I think I will merge it after version 2.0.0 because I don't how much this PR is in conflict with the next release.

File: src/multibody/liegroup/operation-base.hpp [hunk 1]
```diff
@@ -165,20 +166,26 @@ SE3_LIE_GROUP_PUBLIC_INTERFACE_GENERIC(Derived,typename)
...
+                     const Eigen::MatrixBase<JacobianLOut_t>& J0,
+                     const Eigen::MatrixBase<JacobianROut_t>& J1) const
+    PINOCCHIO_DEPRECATED;
+
     /**
      * @brief      Computes the Jacobian of the difference operation.
      *
      * @param[in]  q0    the initial configuration vector.
      * @param[in]  q1    the terminal configuration vector.
+     * @tparam[in] iVar  0 (resp. 1) to get the Jacobian with respect to q0 (resp. q1).
```
Comment (jmirabel) [inline src/multibody/liegroup/operation-base.hpp]: The template parameter is already documented.

**Comment by jmirabel** [comment]
But I have always found `difference(q0,q1)` unclear and I always have to check the doc or the code.

**Comment by jmirabel** [comment]
> I think I will merge it after version 2.0.0 because I don't how much this PR is in conflict with the next release.

I would be happy to put it myself within the beta version. Why isn't there a branch in the main repo containing the content of this release ?
This way, we can actually contribute to the latest point rather than being replied that either:
- you already added it in the next release so the same work was done twice,
- we should wait for your release and someone must make the fix compatible...
There are 4-5 people actively working on Pinocchio and this work would gain from more cooperation.

By that, I do not ask you to merge it if it makes things more complicated.

**Comment by jcarpent** [comment]
I do understand, but I'm not full time on it, and the current version is not passing all the tests, and I don't think that someone else would like to handle it.

File: src/multibody/liegroup/operation-base.hpp (hunk 1 above)
Comment (jcarpent) [reply src/multibody/liegroup/operation-base.hpp]: Ok. I did not see it when I was reading your code.

**Comment by jcarpent** [comment]
Concerning the choice of template naming, I do not think that putting an int is helpful neither.
You will have to in the documentation too, so far and so on.

I was just suggesting a convenient naming, and I suggest to think about it now.
Why not to use something like WITH_RESPECT_TO_FIRST_ARGUMENT, WITH_RESPECT_TO_SECOND_ARGUMENT, etc.

**Comment by jmirabel** [comment]
> I'm not full time on it

I understand this and you already do a lot.

> the current version is not passing all the tests

This is not important for a branch made for elaborating a new release. It is still in beta version so, yes, it is not ready. An appropriate branch name is sufficient to make people understand they should not take it as a stable version. But if you want a patch to be included in the next release, it is the branch you must use.

**Comment by jmirabel** [comment]
> You will have to in the documentation too, so far and so on.

I think I documented it.

> I suggest to think about it now.

Agreed. For clarity, it should be rather short. ARG0, ARG1... ?

**Comment by jcarpent** [comment]
The other more important issue is that this branch is not yet aligned with Pinocchio 1.3.0 and further.
This is a tedious work. I will try to do my best to make it available by the end of the next week as I mentioned in #547.

**Comment by jcarpent** [comment]
Yes, it should be short, but also explaining the argument. WRT_ARG0?

**Comment by jmirabel** [comment]
Is there other places where this problem occurs ?
Otherwise, I can make two functions (like `dDifference_darg0` and  `dDifference_darg1` or  `dDifference_dq0` and  `dDifference_dq1`).

**Comment by gabrielebndn** [comment]
I like dDifference_dq0 and dDifference_dq1. Much better than having a silly enum which is only used for this single method.

**Comment by jmirabel** [comment]
Cherry-picked ont `topic/release-2.0.0` branch. See https://github.com/jcarpent/pinocchio/pull/2

**Pull request closed without merging**